apps/indexer/lib/indexer/fetcher/internal_transaction.ex

```diff
@@ -180,67 +175,81 @@ defmodule Indexer.Fetcher.InternalTransaction do
     end
   end
 
-  defp check_db(num, used_gas, res) do
-    if num != 0 || Decimal.to_integer(used_gas) == 0 do
-      {:ok, res}
-    else
-      {:error, :block_not_indexed_properly}
-    end
-  end
-
-  defp add_block_hash(block_hash, internal_transactions) do
-    Enum.map(internal_transactions, fn a -> Map.put(a, :block_hash, block_hash) end)
-  end
-
   defp fetch_block_internal_transactions_by_transactions(unique_numbers, json_rpc_named_arguments) do
     Enum.reduce(unique_numbers, {:ok, []}, fn
       block_number, {:ok, acc_list} ->
         block = Chain.number_to_any_block(block_number)
 
         block_number
         |> Chain.get_transactions_of_block_number()
-        |> Enum.map(&params(&1))
-        |> case do
-          [] ->
-            {{:ok, []}, 0, block}
-
-          transactions ->
-            try do
-              res = EthereumJSONRPC.fetch_internal_transactions(transactions, json_rpc_named_arguments)
-              {res, Enum.count(transactions), block}
-            catch
-              :exit, error ->
-                {:error, error, block}
-            end
-        end
-        |> case do
-          {{:ok, internal_transactions}, num, {:ok, %{gas_used: used_gas, hash: block_hash}}} ->
-            Logger.debug(fn ->
-              [
-                "Found ",
-                inspect(Enum.count(internal_transactions)),
-                " internal tx for block ",
-                inspect(block_number),
-                " had txs: ",
-                inspect(num),
-                " used gas ",
-                inspect(used_gas)
-              ]
-            end)
-
-            res = add_block_hash(block_hash, internal_transactions) ++ acc_list
-
-            check_db(num, Decimal.new(used_gas), res)
-
-          {error_or_ignore, _, _} ->
-            error_or_ignore
-        end
+        |> extract_transaction_parameters()
+        |> perform_internal_transaction_fetch(block, json_rpc_named_arguments)
+        |> handle_transaction_fetch_results(block_number, acc_list)
 
       _, error_or_ignore ->
         error_or_ignore
     end)
   end
 
+  defp extract_transaction_parameters(transactions) do
+    transactions
+    |> Enum.map(&params(&1))
+  end
+
+  defp params(%{block_number: block_number, hash: hash, index: index, block_hash: block_hash})
```

Comment:
I was thinking about adding a comment why these parameters are needed as it's unclear only looking at this module's code.

Reply:
I can only say that they may be needed by the `EthereumJSONRPC` library to create requests and associate responses with their blocks, but I don't know why it expects this subset of parameters instead of working with the full transaction struct itself.